### PR TITLE
Added a publisher for the standard sensor_msgs::BatteryState message.

### PR DESCRIPTION
**Pull request by allenh1**
The `NAN` values can be replaced later with the correct stats, I just didn't know them.

### CODE REVIEW

**Comment by tfoote** [comment]
@allenh1 It would be better to submit these two very different changes as separate PRs instead of appending to the same PR.

I'd recommend making feature branches and PRing from there instead of your master to avoid this sort of duplication.

File: arduino/opencr_arduino/opencr/libraries/turtlebot3/turtlebot3_waffle/turtlebot3_core/turtlebot3_core_config.h [hunk 1]
```diff
@@ -80,6 +80,7 @@
 #define WAIT_SECOND                      1
 #define CHECK_BUTTON_RELEASED            2
 
+#define ACCEL_FACTOR                     -0.000598  // 2.0 * -9.8 / 32768
```
Comment (tfoote) [inline arduino/opencr_arduino/opencr/libraries/turtlebot3/turtlebot3_waffle/turtlebot3_core/turtlebot3_core_config.h]: Can you link to the document here?

Also you mentioned that we need to add scaling for the other sensors like the gyro?

**Comment by robotpilot** [comment]
I am very grateful for your contribution. Thanks @allenh1 and @tfoote 
But OpenCR's firmware is not easy to change from the point of view of TurtleBot 3's mass production.
I think the sensor_msgs::BatteryState message you sent is a good feature of TB3. 
We will add the feature to the second production period in the future.
Thanks :)

**Comment by allenh1** [comment]
Sure thing! And I'll get the IMU fixed as well (currently, my turtlebots read 15,000+ m/s^2 in the z direction, and several hundred m/s^2 in the x and y direction. I forgot that pushing to my master branch would add it to this pull request for a few moments earlier, and will push remove them from this one and start a new one.

> But OpenCR's firmware is not easy to change from the point of view of TurtleBot 3's mass production.

This is something I talked about with @tfoote the other day. I'm going to look into a way to script this, so that the user could install the rospackage then just call one of the scripts from the ros package to reflash the firmware (thought this would need to be done from a separate machine still, it would be a better long term solution, in my opinion).

You guys have done some great work with this product! As a long term turtlebot fan, I'm definitely enjoying playing with the third generation.

File: arduino/opencr_arduino/opencr/libraries/turtlebot3/turtlebot3_waffle/turtlebot3_core/turtlebot3_core_config.h (hunk 1 above)
Comment (allenh1) [inline arduino/opencr_arduino/opencr/libraries/turtlebot3/turtlebot3_waffle/turtlebot3_core/turtlebot3_core_config.h]: Yes, I'll link the doc here.

**Comment by robotpilot** [comment]
1. ACCEL_FACTOR and BatteryState message
It's a good idea. Thank you.
We will not be able to change it right away, but we will add this feature to our next firmware release as you intended.

2. mass production and factory firmware
Although it may vary from one company to another, in the case of firmware which is closely related to the hardware, the firmware is supplied to the user through various procedures for verification. So it is not easy to change soon. So we are going to add procedures to make it easier to install the latest firmware when users get the product. We take user feedback very seriously. It may be difficult to reflect the proposed functions right away, but we will apply it as soon as possible. :)

**Comment by allenh1** [comment]
@tfoote fixed the git history and whatnot. Should I add the waffle's battery message to this PR?

**Comment by tfoote** [comment]
Adding this to the waffle here too makes sense to me.

**Comment by allenh1** [comment]
Ok, this one should be good to go.

**Comment by robotpilot** [comment]
Merged into branch: feature-from-openrobotics from allenh1-master / https://github.com/ROBOTIS-GIT/OpenCR/commit/7b1f5a5a8e9ff776d562bc5ff5a97a43318a4112
@allenh1, Thank you for your contribution. :)

**Pull request closed without merging**